### PR TITLE
Port from Achilles to ZEN

### PR DESCRIPTION
**Pull request by PhILoXs**
**When merged this pull request will:**
- Port Modules, dialogs, messages to ZEN Framework

### CODE REVIEW

File: addons/zeus/functions/fnc_artyFireMissionHE.sqf [hunk 1]
```diff
@@ -25,44 +25,47 @@ params [
...
-    ]
-] call Ares_fnc_showChooseDialog;
+        ["EDIT", LLSTRING(artillery_ammoAmount), ["4"]],
+        ["COMBO", LLSTRING(artillery_airburst), [[0, 1, 2 , 3], [LLSTRING(artillery_airburst_no), LLSTRING(artillery_airburst_low), LLSTRING(artillery_airburst_med), LLSTRING(artillery_airburst_high)], 0]],
+        ["COMBO", LLSTRING(artillery_delayType), [[0, 1, 2], [LLSTRING(artillery_delay), LLSTRING(artillery_durationWithAmmo), LLSTRING(artillery_durationWithDelay)], 0]],
+        ["EDIT", LLSTRING(artillery_delay), ["1"]],
+        ["EDIT", LLSTRING(artillery_duration), ["60"]]
+    ],
+    {
+        private ["_detonationHight"];
```
Comment (Timi007) [inline addons/zeus/functions/fnc_artyFireMissionHE.sqf:40]:
```suggestion
```

File: addons/zeus/functions/fnc_artyFireMissionHE.sqf [hunk 2]
```diff
@@ -25,44 +25,47 @@ params [
...
+        };
 
-if (_delay < 0 || _duration < 0) exitWith {
-    [LLSTRING(artillery_errorDelayOrHight)] call Ares_fnc_ShowZeusMessage;
-};
-if (_ammoAmount <= 0) exitWith {
-    [LLSTRING(artillery_errorAmount)] call Ares_fnc_ShowZeusMessage;
-};
+        //select airburst hight
+        _detonationHight = [0, 15, 20, 30] select _airburstType;
```
Comment (Timi007) [inline addons/zeus/functions/fnc_artyFireMissionHE.sqf:56]:
```suggestion
        private _detonationHight = [0, 15, 20, 30] select _airburstType;
```

File: addons/zeus/functions/fnc_moduleArtillery.sqf [hunk 3]
```diff
@@ -15,71 +15,74 @@
...
+            ["COMBO", LLSTRING(artillery_centerPositionType), [[0, 1], [LLSTRING(artillery_modulePos), LLSTRING(artillery_customPos)], 0]],
+            ["EDIT", LLSTRING(artillery_centerPositionXPos), ["0000"]],
+            ["EDIT", LLSTRING(artillery_centerPositionYPos), ["0000"]],
+            ["EDIT", LLSTRING(artillery_areaLenght), ["100"]],
+            ["EDIT", LLSTRING(artillery_areaWidth), ["100"]],
+            ["EDIT", LLSTRING(artillery_areaAngle), ["0000"]],
+            ["EDIT", LLSTRING(artillery_timeOnTarget), ["5"]]
+        ],
+        {
+            private ["_position"];
```
Comment (Timi007) [inline addons/zeus/functions/fnc_moduleArtillery.sqf:38]:
```suggestion
            private "_position";
```

File: addons/zeus/functions/fnc_moduleArtillery.sqf [hunk 4]
```diff
@@ -15,71 +15,74 @@
  *
  */
 #include "script_component.hpp"
-
+//CHECK even neccessary?
```
Comment (Timi007) [inline addons/zeus/functions/fnc_moduleArtillery.sqf:18]:
```suggestion
```

Comment (Timi007) [reply addons/zeus/functions/fnc_moduleArtillery.sqf:18]: Is the module called locally on the zeus?
If so then you can remove the checks.

Comment (PhILoXs) [reply addons/zeus/functions/fnc_moduleArtillery.sqf:18]: As of the documentation:
The module function is executed in the unscheduled environment on the client that placed the module.

**Pull request merged**